teps/0081-add-chains-subcommand-to-the-cli.md

```diff
@@ -0,0 +1,337 @@
+{
+  "conditions": [
+    {
+      "type": "Succeeded",
+      "status": "True",
+      "lastTransitionTime": "2021-08-31T15:19:45Z",
+      "reason": "Succeeded",
+      "message": "All Steps have completed executing"
+    }
+  ],
+  "podName": "cache-image-pipelinerun-r-qd6xw-build-trusted-jrj5v-pod-qptxn",
+  "startTime": "2021-08-31T15:19:17Z",
+  "completionTime": "2021-08-31T15:19:45Z",
+  ...
+}
+```
+
+Instead of:
+
+```bash
+TASKRUN=cache-image-pipelinerun-r-qd6xw-build-trusted-jrj5v
+TASKRUN_UID=$(kubectl get taskrun $TASKRUN -o=json | jq -r '.metadata.uid')
+kubectl get taskrun $TASKRUN -o=json \
+  | jq -r ".metadata.annotations[\"chains.tekton.dev/payload-taskrun-$TASKRUN_UID\"]" \
+  |base64 --decode \
+  | jq
+```
+
+The behaviour must be secure by default, therefore verifying the signature when
+extracting the payload. However the ability to disable it should be provided to
+the user via a flag like `-S, --skip-verify`.
+
+#### Use Case 2
+
+Extract the signature from a taskrun:
+
+```bash
+$ tkn chains signature cache-image-pipelinerun-r-qd6xw-build-trusted-jrj5v
+MEUCIQCHS4in92kQW6B+LfnEilpRcOg5akspPh4mQ+tSaEq/jgIgdl83eCyw41xFC8xti6j0/TgzXkKVixfD30yenabWyHU=
+```
+
+Instead of:
+
+```bash
+TASKRUN=cache-image-pipelinerun-r-qd6xw-build-trusted-jrj5v
+TASKRUN_UID=$(kubectl get taskrun $TASKRUN -o=json | jq -r '.metadata.uid')
+kubectl get taskrun $TASKRUN -o=json \
+  | jq -r ".metadata.annotations[\"chains.tekton.dev/signature-taskrun-$TASKRUN_UID\"]"
+```
+
+#### Use Case 3
+
+Change the provenance format:
+
+```bash
+$ tkn chains format in-toto
+```
+
+instead of:
+
+```bash
+ kubectl patch \
+  configmap chains-config \
+  -n tekton-chains \
+  -p='{"data":{"artifacts.taskrun.format": "in-toto"}}'
+```
+
+#### Other Use Case ideas
+
+* Generate and configure `cosign` keys
+* Verify `cosign` signatures
+* Validate `in-toto` layout
+
+## Requirements
+
+There may not be any specific requirements, depending on the commands being
+implemented. See the [Notes/Caveats (optional)](notes-caveats-optional) for more
+details
+
+## Proposal
+
+```shell
+❯ ./tkn chains --help
+Manage Tekton Chains
+
+Usage:
+tkn chains [flags]
+tkn chains [command]
+
+Available Commands:
+  payload     Print a Tekton chains' payload for a specific taskrun
+  signature   Print a Tekton chains' signature for a specific taskrun
+
+Flags:
+  -h, --help       help for chains
+  -C, --no-color   disable coloring (default: false)
+
+Use "tkn chains [command] --help" for more information about a command.
+```
+
+### Notes/Caveats (optional)
+
+Some commands would require adding extra dependencies. For instance we would
+need to either ensure `cosign` is installed on the system or add it as a
+dependency in order to implement the commands having to deal with the `cosign`
+keys and signatures.
+
+However, the Sigstore group is working on a lightweight version of `cosign` that
+could be more easily added as a dependency.
```

Comment:
Note we could also just "bail" when cosign is not there, with a decent message for people to know where to look for installing it.

Reply:
That's a good point. However if we use `cosign` as an external binary instead of the library, I think we will have to store the data into files temporarily to be able to pass it to `cosign`, which is not ideal.

Reply:
I think the library approach is going to be the best one long term.